src/tools/rbd_mirror/image_replayer/BootstrapRequest.cc

```diff
@@ -615,6 +621,15 @@ void BootstrapRequest<I>::handle_get_remote_tags(int r) {
   } else if (reconnect_orphan) {
     dout(20) << ": remote image was demoted/promoted" << dendl;
   } else {
+    if (m_do_resync != nullptr && (*m_local_image_ctx)->journal != nullptr) {
```

Comment:
Should this logic move up so that it covers the non-split brain resync case as well (since you removed the journal resync check within ImageReplayer)?